zerver/views/users.py

```diff
@@ -304,6 +315,7 @@ def add_bot_backend(request, user_profile, full_name_raw=REQ("full_name"), short
                                      base_url=payload_url,
                                      interface=interface_type,
                                      token=random_api_key())
+    notify_created_bot(bot_profile)
```

Comment:
This line puzzles me; that function should be called by `do_create_user` already; is the issue that the Service object is created a bit later and we need to push that detail out in a separate event?